Added new recommendations for GCP governance (#6926)
harness/developer-hub: docs/cloud-cost-management/5-use-ccm-cost-governance/asset-governance/GCP/gcp-recommendations.md

```

**Savings Computed:** The policy identifies a list of resources on which potential savings are calculated by summing up cost of each resource for last 30 days.

**Permissions Required:** 
- Dry Run:
  - `recommender.runServiceCostRecommendations.list`
  - `run.services.list`

- Run Once:
  - `recommender.runServiceCostRecommendations.list`
  - `run.services.list`

---
### Recommendation: gcp-list-unused-bq-datasets
**Description:** List BigQuery datasets that haven't been accessed in the last 7 days.

**Policy Used:** 
```yaml
policies:
  - name: gcp-list-unused-bq-datasets
    resource: gcp.bq-dataset
    description: |
      List BigQuery datasets that haven't been accessed in the last 7 days.
    filters:
      - type: value
        key: lastModifiedTime
        op: less-than
        value_type: age
        value: 7
```

**Savings Computed:** The policy identifies a list of resources on which potential savings are calculated by summing up cost of each resource for last 30 days.

**Permissions Required:** 
- Dry Run:
  - `bigquery.datasets.get`

- Run Once:
  - `bigquery.datasets.get`

---
### Recommendation: gcp-delete-unused-cloud-functions

> **Description:** Delete Cloud Functions that haven't been invoked in the last 30 days to reduce costs.

Reviewer: ```suggestion
**Description:** Delete Cloud Functions that haven't been invoked in the last 7 days to reduce costs.
```

**Policy Used:** 
```yaml
policies:
  - name: gcp-delete-unused-cloud-functions
    resource: gcp.function
    description: >

>       Delete Cloud Functions that haven't been invoked in the last 30 days to

Reviewer: ```suggestion
      Delete Cloud Functions that haven't been invoked in the last 7 days to
```

  - `dataflow.jobs.list`

- Run Once:
  - `dataflow.jobs.list`

---
### Recommendation: gcp-delete-low-utilized-load-balancers
**Description:** Delete all low utilized load balancers where packet count is less than 1000 in last 72 hours

**Policy Used:** 
```yaml
policies:
  - name: gcp-delete-low-utilized-load-balancers
    resource: gcp.loadbalancer-address
    description: >
      Delete all low utilized load balancers where packet count is less than
      1000 in last 72 hours
    filters:
      - type: metrics
        name: compute.googleapis.com/instance/network/received_packets_count
        metric-key: metric.labels.instance_name
        aligner: ALIGN_COUNT
        days: 3
        value: 1000
        op: le
    actions:
      - type: delete
      
```

**Savings Computed:** The policy identifies a list of resources on which potential savings are calculated by summing up cost of each resource for last 30 days.

**Permissions Required:** 
- Dry Run:
  - `monitoring.timeSeries.list`
  - `compute.addresses.list`

- Run Once:
  - `monitoring.timeSeries.list`
  - `compute.addresses.list`
  - `compute.addresses.delete`

---
### Recommendation: gcp-list-unused-redis-instances

> **Description:** List Redis instances with less than 5% CPU utilization in last 7 days

Reviewer 2: ```suggestion
**Description:** List Redis instances with less than 5% CPU utilization over the last 7 days.
```

**Policy Used:** 
```yaml
policies:
  - name: gcp-list-unused-redis-instances
    resource: gcp.redis
    description: List Redis instances with less than 5% CPU utilization in last 7 days
    filters:
      - type: metrics
        name: redis.googleapis.com/stats/cpu_utilization
        metric-key: resource.labels.instance_id
        days: 7
        value: 0.05
        op: lte
      
```

> **Savings Computed:** The policy identifies a list of resources on which potential savings are calculated by summing up cost of each resource for last 30 days.

Reviewer 2: ```suggestion
**Savings Computed:** The policy identifies a list of resources on which potential savings are calculated by summing up cost of each resource for the last 30 days.
```
